### ISSUE TITLE
Pip installation error related to uprading `six`.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#4169]
Installing Ray (both from source and from the latest wheels on S3) leads to

```
Cannot uninstall 'six'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
```

This user has `six==1.10.0`. This changed in https://github.com/ray-project/ray/pull/3675. This looks like a pretty common MacOS problem and will probably cause problems for tons of users, see https://github.com/pypa/pip/issues/3165, so let's downgrade back to what we had before `six >= 1.0.0`.

cc @guoyuhong @justinwyang

**Comment by guoyuhong** [comment]
OK. This is a simple function. I can copy it to `utils.py` and downgrade `six >= 1.0.0`.

**Comment by robertnishihara** [comment]
Fixed by #4180.
